nimbleclient/v1/restclient.py

```diff
@@ -164,81 +168,105 @@ def close_connection(self):
+                    break
+
+                # When limit/pageSize is requested, then return only the limit/pageSize no. of records
+                if ('pageSize' in params and params['pageSize'] is not None) or limit is not None:
                     break
 
-            # Check for errors if any in the response (Treat partial response as an error)
-            if 'messages' in response.json():
-                raise NimOSAPIError(response.json()['messages'])
-
-            # Retrieves as per 'rest_api_row_limit' configuration on array
-            if 'pageSize' in params:
-                return response.json()
-
-            if 'totalRows' not in response.json():
-                return response.json()
-
-            # If startRow and/or endRow is specified, then retrieve records accordingly.
-            # If unspecified, then retrieves all the available records.
-            paginated_data = list()
-            paginated_data.extend(response.json()["data"])
-            total_rows = response.json()['totalRows']
-            retrieved_rows = len(paginated_data)
-            #current_end_row = response.json()['endRow']
-
-            if 'startRow' in params and 'endRow' in params:
-                requested_rows = params['endRow'] - params['startRow']
-            elif 'startRow' not in params and 'endRow' not in params:
-                requested_rows = total_rows # Get all available records
-            else:
+                # When end_row is requested, then return only up to 'end_row' records
+                if 'endRow' in params and params['endRow'] == response['endRow']:
+                    break
+
+                # Set next start row index to continue reading more records
                 if 'startRow' in params:
-                    requested_rows = total_rows - params['startRow']
+                    params['startRow'] += len(response["data"])
                 else:
-                    requested_rows = params['endRow']
-            #print("Total records requested:", requested_rows, ", Retrieved records:", retrieved_rows)
-
-            pending_rows = requested_rows-retrieved_rows
-            params.clear()
-            while pending_rows > 0:
-                params['startRow'] = response.json()['endRow']
-                #print("Retrieved:", retrieved_rows, "Pending:", pending_rows, "Next startRow:", params['startRow'])
-                response = requests.get(
-                    url,
-                    params=params,
-                    headers=self._headers,
-                    verify=False,
-                )
-                records = response.json()["data"][:pending_rows]
-                records_count = len(records)
-                paginated_data.extend(records)
-                pending_rows -= records_count
-                retrieved_rows += records_count
-                # TODO: Handle change in the total no. of rows during this large operation
+                    params['startRow'] = len(response["data"])
```

Comment:
After changing to limit / from_id, we no longer need the logic in lines 231-261

Reply:
As agreed, we can take out the startRow/endRow support and fix this